vcd/resource_vcd_nsxt_network_dhcp.go

```diff
@@ -0,0 +1,263 @@
+package vcd
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var nsxtDhcpPoolSetSchema = &schema.Resource{
+	Schema: map[string]*schema.Schema{
+		"start_address": &schema.Schema{
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "Start address of DHCP pool IP range",
+		},
+		"end_address": &schema.Schema{
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "End address of DHCP pool IP range",
+		},
+	},
+}
+
+func resourceVcdOpenAPIDhcp() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceVcdOpenAPIDhcpCreate,
+		ReadContext:   resourceVcdOpenAPIDhcpRead,
+		UpdateContext: resourceVcdOpenAPIDhcpUpdate,
+		DeleteContext: resourceVcdOpenAPIDhcpDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceVcdOpenAPIDhcpImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Description: "The name of organization to use, optional if defined at provider " +
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"vdc": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The name of VDC to use, optional if defined at provider level",
+			},
+
+			"org_network_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Parent Org VDC network ID",
+			},
+			"pool": &schema.Schema{
+				Type:        schema.TypeSet,
+				Required:    true,
+				Description: "IP ranges used for DHCP pool allocation in the network",
+				Elem:        nsxtDhcpPoolSetSchema,
+			},
+		},
+	}
+}
+
+// resourceVcdOpenAPIDhcpCreate
+func resourceVcdOpenAPIDhcpCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

Comment:
I noticed a naming consistency issue: some functions are name "...OpenApi..." and some "...OpenAPI..." - shouldn't all be named "...Api..."?

Reply:
My bad. Replaced what I could find with grep (looks like this is only this PR). Some linters advise for upper cased abbreviations and this hooked me.